tests/unit/syncModel/validation/languageSchema.test.ts

```diff
@@ -0,0 +1,135 @@
+import { describe, expect, test } from "vitest";
+
+import { LanguageSchema } from "../../../../src/modules/sync/validation/entitySchema.js";
+
+describe("LanguageSchema Tests", () => {
+  test("Valid object with is_default true", () => {
+    const validData = {
+      name: "English",
+      codename: "en",
+      is_active: true,
+      is_default: true,
+    };
+
+    const result = LanguageSchema.safeParse(validData);
+    expect(result.success).toBe(true);
+  });
+
+  test("Valid object with is_default false and valid fallback_language", () => {
+    const validData = {
+      name: "French",
+      codename: "fr",
+      is_active: true,
+      is_default: false,
+      fallback_language: {
+        codename: "en",
+      },
+    };
+
+    const result = LanguageSchema.safeParse(validData);
+    expect(result.success).toBe(true);
+  });
+
+  test("Invalid object with is_default true and fallback_language provided", () => {
+    const invalidData = {
+      name: "English",
+      codename: "en",
+      is_active: true,
+      is_default: true,
+      fallback_language: {
+        codename: "en",
+      },
+    };
+
+    const result = LanguageSchema.safeParse(invalidData);
+    expect(result.success).toBe(false);
+  });
+
+  test("Invalid object with is_default false and missing fallback_language", () => {
```

Comment:
Although I like the enthusiasm with tests, I would rely on TS forcing us to create the schemas properly and then `zod` to behave properly with the right schemas. You didn't write tests on all the schemas anyway. I think that one valid and one invalid test per schema is more than enough and instead we can cover all the schemas.